test/api/test_gpu.cpp

```diff
@@ -45,27 +46,68 @@ TEST_CASE(load_and_run)
     {
         pp.add(name, migraphx::argument::generate(param_shapes[name]));
     }
+
     auto outputs = p.eval(pp);
     CHECK(shapes_before.size() == outputs.size());
     CHECK(bool{shapes_before.front() == outputs.front().get_shape()});
 }
 
+using hip_ptr = MIGRAPHX_MANAGE_PTR(void, hipFree);
+
+hip_ptr get_hip_host_buffer(size_t size)
+{
+    void* ptr;
+    auto err = hipHostMalloc(&ptr, size);
+    if(err != hipSuccess)
+    {
+        EXPECT(false);
+    }
+    return hip_ptr{ptr};
+}
+
+hip_ptr get_hip_buffer(size_t size)
+{
+    void* ptr;
+    auto err = hipMalloc(&ptr, size);
+    if(err != hipSuccess)
+    {
+        EXPECT(false);
+    }
+    return hip_ptr{ptr};
+}
+
 TEST_CASE(load_and_run_async)
 {
     auto p             = migraphx::parse_onnx("conv_relu_maxpool_test.onnx");
     auto shapes_before = p.get_output_shapes();
     migraphx::compile_options options;
-    options.set_offload_copy();
+    options.set_offload_copy(false);
     p.compile(migraphx::target("gpu"), options);
     auto shapes_after = p.get_output_shapes();
     CHECK(shapes_before.size() == 1);
     CHECK(shapes_before.size() == shapes_after.size());
     CHECK(bool{shapes_before.front() == shapes_after.front()});
     migraphx::program_parameters pp;
     auto param_shapes = p.get_parameter_shapes();
+
+    std::vector<hip_ptr> host_buffs;
+    std::vector<hip_ptr> buffs;
+    std::vector<migraphx::argument> args;
     for(auto&& name : param_shapes.names())
     {
-        pp.add(name, migraphx::argument::generate(param_shapes[name]));
+        args.push_back(migraphx::argument::generate(param_shapes[name]));
+        buffs.push_back(get_hip_buffer(args.rbegin()->get_shape().bytes()));
+        host_buffs.push_back(get_hip_buffer(args.rbegin()->get_shape().bytes()));
```

Comment:
Doesnt' this `host_buffs` needs to point to `args` first before starting copy to GPU ?

Reply:
Ah got it. Ill fix that. Printing it out looks like I was getting all zeros.

Reply:
Done